src/main/java/org/openstreetmap/atlas/geography/atlas/change/FeatureChangeMergers.java

```diff
@@ -0,0 +1,321 @@
+package org.openstreetmap.atlas.geography.atlas.change;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.geography.atlas.builder.RelationBean;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasEntity;
+import org.openstreetmap.atlas.utilities.collections.Maps;
+import org.openstreetmap.atlas.utilities.collections.Sets;
+import org.openstreetmap.atlas.utilities.function.TernaryOperator;
+
+/**
+ * A utility class to store the various merger operations utilized by {@link FeatureChange}.
+ *
+ * @author lcram
+ */
+public final class FeatureChangeMergers
+{
+    /**
+     * A bean class to store the merged before and after members. This is useful as a return type
+     * for the member merger, which needs to correctly merge the before and after entity view of
+     * each {@link FeatureChange}.
+     *
+     * @author lcram
+     * @param <M>
+     *            the member type
+     */
+    public static class MergedMemberBean<M>
+    {
+        private final M beforeMemberMerged;
+        private final M afterMemberMerged;
+
+        public MergedMemberBean(final M before, final M after)
+        {
+            this.beforeMemberMerged = before;
+            this.afterMemberMerged = after;
+        }
+
+        public M getMergedAfterMember()
+        {
+            return this.afterMemberMerged;
+        }
+
+        public M getMergedBeforeMember()
+        {
+            return this.beforeMemberMerged;
+        }
+    }
+
+    static final BinaryOperator<Map<String, String>> tagMerger = Maps::withMaps;
+
+    static final BinaryOperator<Set<Long>> directReferenceMerger = Sets::withSets;
+
+    static final BinaryOperator<SortedSet<Long>> directReferenceMergerSorted = Sets::withSortedSets;
+
+    static final BinaryOperator<SortedSet<Long>> directReferenceMergerLooseSorted = (left,
+            right) -> Sets.withSortedSets(false, left, right);
+
+    static final BinaryOperator<Set<Long>> directReferenceMergerLoose = (left, right) -> Sets
+            .withSets(false, left, right);
+
+    static final BinaryOperator<RelationBean> relationBeanMerger = RelationBean::merge;
+
+    static final TernaryOperator<SortedSet<Long>> diffBasedIntegerSortedSetMerger = (before,
```

Comment:
An example of how a diff based merger can work. This one merges the in/out edge identifier sets. This logic may be generalizable.

Reply:
I am leaving them because we fall back on them in cases where no beforeView is provided.